src/KeraLua/NativeMethods.cs

```diff
@@ -28,6 +27,12 @@ static NativeMethods ()
 
 #endif
 
+		[DllImport(LIBNAME, CallingConvention = CallingConvention.Cdecl, EntryPoint = "lua_pushcclosure")]
+		internal static extern void LuaPushCClosure(IntPtr luaState, LuaNativeFunction fn, int n);

```

Comment:
Why not use the `pushstdcallcfunction` ? 😉 
The idea here is to interop with .NET `GetFunctionPointerForDelegate` will return a pointer only called using the `__stdcall` calling convention.